axum-core/src/macros.rs

```diff
@@ -1,4 +1,79 @@
-macro_rules! define_rejection {
+/// Private API.
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __log_rejection {
+    (
+        rejection_type = $ty:ident,
+        body_text = $body_text:expr,
+        status = $status:expr,
+    ) => {
+        #[cfg(feature = "tracing")]
+        {
+            tracing::event!(
+                target: "axum::rejection",
+                tracing::Level::TRACE,
+                status = $status.as_u16(),
+                body = %$body_text,
```

Comment:
Good question 🤔 it's not needed. I've removed it.

Reply:
Wut? Docs show that the impl exists: https://docs.rs/tracing/latest/tracing/trait.Value.html#impl-Value-for-String

Also, I think the default formatting for `str` is equivalent to the `?debug` formatting rather than `%display` formatting.